### PR TITLE
Added service config auto rollouts e2e test for GCE

### PR DESCRIPTION
**Pull request by mangchiandjjoe**

### CODE REVIEW

File: test/bookstore/gce/e2e.sh [hunk 1]
```diff
@@ -81,18 +81,72 @@ run cat "${VM_STARTUP_SCRIPT}"
...
   --boot-disk-size "100GB" \
-  --metadata endpoints-service-name="${ESP_SERVICE}",endpoints-service-version="${ESP_SERVICE_VERSION}" \
+  --metadata "${METADATA}" \
   --metadata-from-file startup-script="${VM_STARTUP_SCRIPT}"
 
 run retry -n 3 get_host_ip "${INSTANCE_NAME}"
 HOST="http://${HOST_INTERNAL_IP}:8080"
 
+CHECK_RETRY_COUNT=0
+until [ $CHECK_RETRY_COUNT -ge 10 ]
```
Comment (qiwzhang) [inline test/bookstore/gce/e2e.sh]: can we use "retry" function?

File: test/bookstore/gce/e2e.sh [hunk 2]
```diff
@@ -81,18 +81,72 @@ run cat "${VM_STARTUP_SCRIPT}"
...
+  exit 1
+fi
+
+
+if [ "${ESP_ROLLOUT_STRATEGY}" == "managed" ]; then
+  # Deploy new service config
+  create_service "${ESP_SERVICE}" swagger.json
+
+  CHECK_RETRY_COUNT=0
+  until [ $CHECK_RETRY_COUNT -ge 10 ]
```
Comment (qiwzhang) [inline test/bookstore/gce/e2e.sh]: same here, try to use "retry" function,   code is more readable.

File: test/bookstore/gce/e2e.sh [hunk 3]
```diff
@@ -81,18 +81,72 @@ run cat "${VM_STARTUP_SCRIPT}"
...
+
+if [ "$ESP_SERVICE_VERSION" != "$ESP_CURRENT_ROLLOUTS" ]; then
+  echo "Rollouts server config update failed"
+  exit 1
+fi
+
+
+if [ "${ESP_ROLLOUT_STRATEGY}" == "managed" ]; then
+  # Deploy new service config
+  create_service "${ESP_SERVICE}" swagger.json
```
Comment (qiwzhang) [inline test/bookstore/gce/e2e.sh]: please add comment,
 create_service will set ESP_SERVICE_VERSION to the deployed new config id

File: test/bookstore/gce/e2e.sh (hunk 2 above)
Comment (mangchiandjjoe) [inline test/bookstore/gce/e2e.sh]: Done

File: test/bookstore/gce/e2e.sh (hunk 3 above)
Comment (mangchiandjjoe) [inline test/bookstore/gce/e2e.sh]: Done

File: test/bookstore/gce/e2e.sh (hunk 1 above)
Comment (mangchiandjjoe) [inline test/bookstore/gce/e2e.sh]: Done

**Comment by mangchiandjjoe** [comment]
Created a new PR https://github.com/cloudendpoints/esp/pull/232

**Pull request closed without merging**